File: src/org/ggp/base/util/gdl/grammar/GdlOr.java
package org.ggp.base.util.gdl.grammar;

import java.util.List;

class GdlOr(GdlLiteral):
{

    private final List<GdlLiteral> disjuncts;
    private transient Boolean ground;

    GdlOr(List<GdlLiteral> disjuncts)
	{
        this.disjuncts = disjuncts;
        ground = null;

    def int arity()
	{
        return disjuncts.size();

    private boolean computeGround()
	{
        for (GdlLiteral literal : disjuncts)
		{
            if (!literal.isGround())
			{
                return false;

        return true;

    def GdlLiteral get(int index)
	{
        return disjuncts.get(index);

    def boolean isGround()
	{
        if (ground == null)
		{
            ground = computeGround();

        return ground;

    def String toString()
	{
        StringBuilder sb = new StringBuilder();

        sb.append("( or ");
        for (GdlLiteral literal : disjuncts)
		{
            sb.append(literal + " ");
        sb.append(")");

        return sb.toString();

File: ggpy/cruft/autocode/GdlOr.py
#!/usr/bin/env python
""" generated source for module GdlOr """
# package: org.ggp.base.util.gdl.grammar
import java.util.List

@SuppressWarnings("serial")
class GdlOr(GdlLiteral):
    """ generated source for class GdlOr """
    disjuncts = List()
    ground = bool()

    def __init__(self, disjuncts):
        """ generated source for method __init__ """
        super(GdlOr, self).__init__()
        self.disjuncts = disjuncts
        self.ground = None

    def arity(self):
        """ generated source for method arity """
        return len(self.disjuncts)

    def computeGround(self):
        """ generated source for method computeGround """
        for literal in disjuncts:
            if not literal.isGround():
                return False
        return True

    def get(self, index):
        """ generated source for method get """
        return self.disjuncts.get(index)

    def isGround(self):
        """ generated source for method isGround """
        if self.ground == None:
            self.ground = self.computeGround()
        return self.ground

    def __str__(self):
        """ generated source for method toString """
        sb = StringBuilder()
        sb.append("( or ")
        for literal in disjuncts:
            sb.append(literal + " ")
        sb.append(")")
        return sb.__str__()

